src/transformers/testing_utils.py

```diff
@@ -3508,3 +3887,313 @@ def torchrun(script: str, nproc_per_node: int, is_torchrun: bool = True, env: Op
             _ = subprocess.run(cmd, capture_output=True, env=env, text=True, check=True)
         except subprocess.CalledProcessError as e:
             raise Exception(f"The following error was captured: {e.stderr}")
+
+
+def _format_tensor(t, indent_level=0, sci_mode=None):
```

Comment:
This and `_format_py_obj` below are just for formatting the values on the file or terminal.